Core/AppDeepLinks.swift

```diff
@@ -37,12 +37,15 @@ public struct AppDeepLinks {
     
     public static func isQuickLink(url: URL) -> Bool {
         if let scheme = url.scheme {
-            return AppDeepLinks.quickLink.contains(scheme)
+            return AppDeepLinks.quickLink.lowercased().contains(scheme.lowercased())
         }
         return false
     }
     
     public static func query(fromQuickLink url: URL) -> String {
-        return url.absoluteString.replacingOccurrences(of: quickLink, with: "")
+        let components = NSURLComponents.init(url: url, resolvingAgainstBaseURL: true)!
+        components.scheme = nil // remove scheme
+        let string = String(components.url!.absoluteString.dropFirst(2)) // strip off '//' prefix
+        return string
     }
```

Comment:
Unfortunately this code breaks the link we have to our about page.  If a link comes in with a scheme then it appears to mangle it slightly, e.g. `https://duckduckgo.com/about` becomes `https//duckduckgo.com/about` (ie no colon after https) and then it doesn't go to the expected page.

Reply:
@brindy Good catch. There's a few solutions to this I could see. I think the right thing to do is to change the value of `aboutLink` to match the expectations of the new implementation of `-queryFromQuickLink`. The two I'm thinking of are

- Changing `aboutLink` to `ddgQuickLink://duckduckgo.com/about`, which implicitly transforms it to `http` later on.
- Changing `aboutLink` to `ddgQuickLink://https%3A%2F%2Fduckduckgo.com%2Fabout`. Having the host be a fully URL encoded URL makes it so that the scheme isn't dropped, which preserves the `https`.

Which would you prefer?

Reply:
Ah wait, just kidding, URL encoding doesn't seem to work. I'd opt for the first solution here, if there's something else you'd prefer please let me know.

Reply:
I believe this is resolved in https://github.com/duckduckgo/iOS/pull/274/commits/a06d9c1f8ced82ce56657162d2a5b721aedd30d2.